### PR TITLE
Feature/invoices

### PR DESCRIPTION
**Pull request by diegosteiner**

### CODE REVIEW

File: app/models/esr_service.rb [hunk 1]
```diff
@@ -20,4 +20,22 @@ def format_ref(ref)
 
     ref.reverse.chars.in_groups_of(5).reverse.map { |group| group.reverse.join }.join(' ')
   end
+
+  def code_line(esr_participant_nr: 'XXX-XXXXX-X', ref:, amount:, esr_mode: '01')
```
Comment (diegosteiner) [inline app/models/esr_service.rb:24]: TODO: Use ENV? I18n? as defaults

File: app/services/limesurvey_service.rb [hunk 2]
```diff
@@ -4,14 +4,18 @@ class LimesurveyService
   BASEURL = 'https://limesurvey.bula21.ch/index.php'
   ADMIN_REMOTECONTROL_URL = URI.parse(BASEURL + '/admin/remotecontrol')
 
-  attr_reader :session_key
   def initialize(username = ENV['LIMESURVEY_USERNAME'],
                  password = ENV['LIMESURVEY_PASSWORD'],
                  survey_id = ENV['LIMESURVEY_SURVEY_ID'])
-    @session_key = fetch_session_key(username, password)
```
Comment (diegosteiner) [inline app/services/limesurvey_service.rb:11]: fetch_session_key always makes a request. We don't need the session in every case, so it should be created lazily

File: app/models/esr_service.rb [hunk 3]
```diff
@@ -20,4 +20,22 @@ def format_ref(ref)
...
+    code = {
+      esr_mode: esr_mode,
+      amount: amount * 100,
```
[the remark this answers is not in the document]
Comment (sihu) [reply app/models/esr_service.rb:27]: we could extract this, so we always use amount_in_cents and it's not forgotten

File: app/services/invoice_builder.rb [hunk 4]
```diff
@@ -1,32 +1,39 @@
 # frozen_string_literal: true
 
 class InvoiceBuilder
```
Comment (sihu) [inline app/services/invoice_builder.rb:3]: this class is not yet used anywhere right? Could we add a button to be able to generate some invoices to test?

Comment (diegosteiner) [reply app/services/invoice_builder.rb:3]: @lukaseisenring wird die Rechnungen per Konsole generieren, dann braucht es den Knopf nicht (noch nicht)

File: app/models/esr_service.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (diegosteiner) [reply app/models/esr_service.rb:27]: Agreed

File: .ruby-version [hunk 5]
```diff
@@ -1 +0,0 @@
-2.6.5
```
[the remark this answers is not in the document]
Comment (diegosteiner) [reply .ruby-version:1]: Hmm good question. It's only needed when developing outside of a container. I'll keep it in

**Pull request merged**